nav2_route/src/plugins/graph_file_loaders/geojson_graph_file_loader.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2023 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <filesystem>
+#include <memory>
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include "nav2_route/plugins/graph_file_loaders/geojson_graph_file_loader.hpp"
+
+namespace nav2_route
+{
+
+void GeoJsonGraphFileLoader::configure(
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr node)
+{
+  RCLCPP_INFO(node->get_logger(), "Configuring geojson graph file loader");
+  logger_ = node->get_logger();
+}
+
+bool GeoJsonGraphFileLoader::loadGraphFromFile(
+  Graph & graph, GraphToIDMap & graph_to_id_map, std::string filepath)
+{
+  if (!fileExists(filepath)) {
+    RCLCPP_ERROR(logger_, "Failed to find file at %s", filepath.c_str());
+    return false;
+  }
+
+  std::ifstream graph_file(filepath);
+
+  Json graph_geojson;
+  graph_file >> graph_geojson;
+
+  auto features = graph_geojson["features"];
+  std::vector<Json> nodes;
+  std::vector<Json> edges;
+  getGraphElements(features, nodes, edges);
+
+  if (nodes.empty()) {
+    RCLCPP_ERROR(logger_, "No nodes were found in %s", filepath.c_str());
+    return false;
+  }
+
+  if (edges.empty()) {
+    RCLCPP_ERROR(logger_, "No edges were found in %s", filepath.c_str());
+    return false;
+  }
+
+  graph.resize(nodes.size());
+  addNodesToGraph(graph, graph_to_id_map, nodes);
+  addEdgesToGraph(graph, graph_to_id_map, edges);
+
+  return true;
+}
+
+bool GeoJsonGraphFileLoader::fileExists(const std::string & filepath)
+{
+  return std::filesystem::exists(filepath);
```

Comment:
Also check `std::ifstream xml_file(filename); if (!xml_file.good()) {return false;}` to check that the file is well formed in XML Format standards. This catches if there's syntactical errors. 

You could actually store that `ifstream` object for use later in the main function. I don't know how heavy that is to reconstruct if its meaningful.